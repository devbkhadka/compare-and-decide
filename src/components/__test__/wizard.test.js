import React from 'react'
import ReactDOM from 'react-dom'
import { render, fireEvent } from '@testing-library/react'

import TabbedWizard from '../wizard'
import TabPanel from '../tab_panel'

describe("Test Wizard Component", ()=>{
    it("should render tabs correctly", ()=> {
        const wizard = render(<TabbedWizard tabs={['tab1', 'tab2']}></TabbedWizard>)
        
        wizard.getByText('tab1')
        wizard.getByText('tab2')
    })

    it("should show first panel when rendered", ()=> {
        const wizard = render(
            <TabbedWizard tabs={['tab1', 'tab2']}>
                <TabPanel>Tab1 content</TabPanel>
                <TabPanel>Tab2 content</TabPanel>
            </TabbedWizard>
        )

        wizard.getByText('Tab1 content')

        expect(wizard.queryByText('Tab2 content')).toBe(null)
    })

    it("clicking on tab should change the tap", ()=> {
        const wizard = render(
            <TabbedWizard tabs={['tab1', 'tab2']}>
                <TabPanel>Tab1 content</TabPanel>
                <TabPanel>Tab2 content</TabPanel>
            </TabbedWizard>
        )

        fireEvent.click(wizard.getByTestId('tab-1'))

        wizard.getByText('Tab2 content')
        
    })

    test('Wizard snapshot is not changed', ()=>{
        const wizard = render(
            <TabbedWizard tabs={['tab1', 'tab2']}>
                <TabPanel>Tab1 content</TabPanel>
                <TabPanel>Tab2 content</TabPanel>
            </TabbedWizard>
        )

        expect(wizard.baseElement).toMatchSnapshot()
    })

})

