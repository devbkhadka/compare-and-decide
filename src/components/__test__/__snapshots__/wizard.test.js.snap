// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Wizard Component Wizard snapshot is not changed 1`] = `
<body>
  <div>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiTabs-root tabs"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              class="MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
                data-testid="tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  tab1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                data-testid="tab-1"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  tab2
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-251 PrivateTabIndicator-colorSecondary-253 MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="tap-panel"
        >
          Tab1 content
        </div>
        <div
          class="tap-panel"
        />
      </div>
    </div>
  </div>
</body>
`;
